### PR TITLE
Centralized JWT Provider

### PR DESCRIPTION
**Pull request by nikhil-rajesh**
## Description
- Wrote a centralised JWT provider and modified all services to use the same
- Migrated JWT algorithm from symmetric to asymmetric

### Testing
- deployed and tested the service locally. 
- Ran the postman collection

### Documentation
- Updated postman requests

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged
- [x] I have documented any changes if required in the [docs](docs).

### CODE REVIEW

File: services/identity/src/main/java/com/crapi/config/JwtProvider.java [hunk 1]
```diff
@@ -37,15 +36,13 @@ public class JwtProvider {
   private int jwtExpiration;
 
   /**
-   * @param authentication
+   * @param user
    * @return generated token with expire date
    */
-  public String generateJwtToken(Authentication authentication) {
-
-    UserPrinciple userPrincipal = (UserPrinciple) authentication.getPrincipal();
```
[the remark this answers is not in the document]
Comment (nikhil-rajesh) [reply services/identity/src/main/java/com/crapi/config/JwtProvider.java:45]: generateJwtToken() was replicated in another file for the verify-email-token endpoint. I wanted to remove that and centralise it. 

We are still using the Spring Framework Authentication to authenticate right? I just fetched the user from db and passed if authentication was successful.
https://github.com/OWASP/crAPI/pull/148/files#diff-9d65eddcd9f473943145c6819204fd245e4143840c9669a0470dd844389ff57b

File: services/identity/src/main/java/com/crapi/controller/AuthController.java [hunk 2]
```diff
@@ -80,6 +80,24 @@ public ResponseEntity<CRAPIResponse> registerUser(@Valid @RequestBody SignUpForm
...
+   * @return success and failure message after token authentication.
+   */
+  @PostMapping("/verify")
+  public ResponseEntity<CRAPIResponse> verifyJwtToken(
+      @Valid @RequestBody JwtTokenForm verifyTokenRequest) {
+    // Creating user's account
+    CRAPIResponse verifyTokenResponse = userService.verifyJwtToken(verifyTokenRequest.getToken());
+    if (verifyTokenResponse != null && verifyTokenResponse.getStatus() == 200) {
+      return ResponseEntity.status(HttpStatus.OK).body(verifyTokenResponse);
+    } else if (verifyTokenResponse != null && verifyTokenResponse.getStatus() == 400) {
```
Comment (piyushroshan) [inline services/identity/src/main/java/com/crapi/controller/AuthController.java:94]: Are we not handling 401/403 here?

File: services/identity/src/main/java/com/crapi/service/Impl/UserServiceImpl.java [hunk 3]
```diff
@@ -124,6 +114,21 @@ public JwtResponse authenticateUserLogin(LoginForm loginForm)
...
+   */
+  @Transactional
+  @Override
+  public CRAPIResponse verifyJwtToken(String token) {
+    boolean validToken = jwtProvider.validateJwtToken(token);
+    if (validToken) {
+      return new CRAPIResponse(UserMessage.VALID_JWT_TOKEN, 200);
+    }
+
+    return new CRAPIResponse(UserMessage.INVALID_JWT_TOKEN, 400);
```
Comment (piyushroshan) [inline services/identity/src/main/java/com/crapi/service/Impl/UserServiceImpl.java:129]: If token exists, it should be 401 Unauthorized

Comment (nikhil-rajesh) [reply services/identity/src/main/java/com/crapi/service/Impl/UserServiceImpl.java]: Updated.

File: services/identity/src/main/java/com/crapi/controller/AuthController.java (hunk 2 above)
Comment (nikhil-rajesh) [reply services/identity/src/main/java/com/crapi/controller/AuthController.java]: Updated.

**Comment by piyushroshan** [comment]
Are we extracting the public key from private to expose for . well-known

**Comment by nikhil-rajesh** [comment]
> Are we extracting the public key from private to expose for . well-known

Yes, we dont have to keep it separately right? 
image

**Pull request merged**